sig-provider/sig-provider-server/src/service.rs

```diff
@@ -40,8 +39,10 @@ impl AbiService for Service {
         request: tonic::Request<GetFunctionAbiRequest>,
     ) -> Result<tonic::Response<GetFunctionAbiResponse>, tonic::Status> {
         let request = request.into_inner();
+        let bytes = hex::decode(request.tx_input)
+            .map_err(|e| tonic::Status::invalid_argument(e.to_string()))?;
```

Comment:
`hex::decode` cannot parse hex strings which start with "0x" prefix. I suggest to copy `ethers_core::types::Bytes` implementation which allows both options. Later we will extract that into a separate lib crate, so that it could be reused in all our services, but before that I suggest to copy it inside your service crate